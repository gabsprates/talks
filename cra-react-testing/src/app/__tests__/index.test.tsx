import React from "react";
import { App } from "../index";
import { render, fireEvent } from "@testing-library/react";

jest.mock("../../services/github");

describe("Component: App", () => {
  it("should render without errors", () => {
    const wrapper = render(<App />);
    expect(wrapper.container.outerHTML).toMatchSnapshot();
  });

  it("should open dev details and clean after click the button", () => {
    const wrapper = render(<App />);

    let button = wrapper.queryByText("clear user");
    expect(button).toBeNull();

    let user = wrapper.getAllByRole("item-user")[0];
    fireEvent.click(user);

    expect(wrapper.getByText("dev details"));

    user = wrapper.getAllByRole("item-user")[1];
    fireEvent.click(user);

    expect(wrapper.getByText("dev details"));

    button = wrapper.getByText("clear user");
    fireEvent.click(button);

    expect(wrapper.queryByText("dev details")).toBeNull();
    expect(wrapper.queryByText("clear user")).toBeNull();
  });
});
