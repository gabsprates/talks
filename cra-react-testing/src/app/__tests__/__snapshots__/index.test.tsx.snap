// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: App should render without errors 1`] = `"<div><section class=\\"app\\"><h1>\\"Zone Of Front-Enders\\"</h1><div class=\\"columns\\"><aside><ul><li><div class=\\"user\\"><strong class=\\"name\\">Albertinin Santos</strong><div class=\\"github\\">albertininm</div></div></li><li><div class=\\"user\\"><strong class=\\"name\\">Bruno Carvalho</strong><div class=\\"github\\">bcarvalho89</div></div></li><li><div class=\\"user\\"><strong class=\\"name\\">Gabriel Prates</strong><div class=\\"github\\">gabsprates</div></div></li><li><div class=\\"user\\"><strong class=\\"name\\">Jared Gorski</strong><div class=\\"github\\">jaredgorski</div></div></li><li><div class=\\"user\\"><strong class=\\"name\\">Mois√©s Menezes</strong><div class=\\"github\\">mgmenezes</div></div></li><li><div class=\\"user\\"><strong class=\\"name\\">Ygor Costa</strong><div class=\\"github\\">ygorcosta</div></div></li><li><div class=\\"user\\"><strong class=\\"name\\">Mateus Castro</strong><div class=\\"github\\">MateusCastro</div></div></li></ul></aside></div></section></div>"`;
